lib/history/sqlite/event.go

```diff
@@ -0,0 +1,98 @@
+package sqlite
+
+import (
+	"database/sql"
+	"time"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/satellite/lib/history"
+
+	"github.com/gravitational/trace"
+)
+
+// sqlEvent defines an sql event row.
+type sqlEvent struct {
+	ID        int            `db:"id"`
+	Timestamp time.Time      `db:"timestamp"`
+	EventType string         `db:"type"`
+	Node      sql.NullString `db:"node"`
+	Probe     sql.NullString `db:"probe"`
+	Old       sql.NullString `db:"oldState"`
+	New       sql.NullString `db:"newState"`
+}
+
+// newSQLEvent constructs a new sqlEvent from the provided TimelineEvent.
+func newSQLEvent(event *pb.TimelineEvent) (row sqlEvent, err error) {
+	row = sqlEvent{Timestamp: event.GetTimestamp().ToTime()}
+	switch t := event.GetData().(type) {
+	case *pb.TimelineEvent_ClusterRecovered:
+		row.EventType = clusterRecoveredType
+	case *pb.TimelineEvent_ClusterDegraded:
+		row.EventType = clusterDegradedType
+	case *pb.TimelineEvent_NodeAdded:
+		e := event.GetNodeAdded()
+		row.EventType = nodeAddedType
+		row.Node = sql.NullString{String: e.GetNode(), Valid: true}
+	case *pb.TimelineEvent_NodeRemoved:
+		e := event.GetNodeRemoved()
+		row.EventType = nodeRemovedType
+		row.Node = sql.NullString{String: e.GetNode(), Valid: true}
+	case *pb.TimelineEvent_NodeRecovered:
+		e := event.GetNodeRecovered()
+		row.EventType = nodeRecoveredType
+		row.Node = sql.NullString{String: e.GetNode(), Valid: true}
+	case *pb.TimelineEvent_NodeDegraded:
+		e := event.GetNodeDegraded()
+		row.EventType = nodeDegradedType
+		row.Node = sql.NullString{String: e.GetNode(), Valid: true}
+	case *pb.TimelineEvent_ProbeSucceeded:
+		e := event.GetProbeSucceeded()
+		row.EventType = probeSucceededType
+		row.Node = sql.NullString{String: e.GetNode(), Valid: true}
+		row.Probe = sql.NullString{String: e.GetProbe(), Valid: true}
+	case *pb.TimelineEvent_ProbeFailed:
+		e := event.GetProbeFailed()
+		row.EventType = probeFailedType
+		row.Node = sql.NullString{String: e.GetNode(), Valid: true}
+		row.Probe = sql.NullString{String: e.GetProbe(), Valid: true}
+	default:
+		return row, trace.BadParameter("unknown event type %T", t)
+	}
+	return row, nil
+}
+
+// toArgs returns sqlEvent as a list of arguments.
+func (e *sqlEvent) toArgs() (args []interface{}) {
```

Comment:
same here - invert the direction since it does not tell me anything about where and for each purpose the list is. Instead, use a descriptive and focused interface (like `ValueInserter`) which would accept the DB instance.